dev/lint/run_clang_format.sh

```diff
@@ -18,8 +18,16 @@
 
 set -ex
 
+reset_owner() {
+  if [ $OWNER ]
+  then
+    chown -R $OWNER /arrow/cpp
+  fi
+}
+
 mkdir -p /build/lint
 pushd /build/lint
   cmake -GNinja /arrow/cpp
+  trap reset_owner EXIT
```

Comment:
This seems fishy. I'd say this is a problem with the docker setup. You can also just run `check-format` for the same purpose without the side effect.

Reply:
I've tried setting [`user: ${UID}`](https://docs.docker.com/compose/compose-file/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir) in docker-compose.yml, but the artifacts produced by `docker-compose build cpp` etc are owned by root. I don't know how to pass my uid to `docker-compose build`.

I'm not sure what you mean with `clang-format`; that's an assertion whereas `docker-compose.yml` on master runs [`ninja format`](https://github.com/apache/arrow/blob/master/dev/lint/run_clang_format.sh#L24), which modifies files in place.

Reply:
@fsaintjacques How's this? I create a [user:group](https://github.com/apache/arrow/pull/4293/files#diff-28781500a62df443db3546cfdca67732R24) with the same ids as the host user, then `su` run the linters [with that](https://github.com/apache/arrow/pull/4293/files#diff-2bb746b0d6cff075e06c0bef65542e4cR26)